main.py

```diff
@@ -33,7 +34,11 @@ def create_menu():
 
 def add_book():
     new_book = ui.get_book_info()
-    new_book.save()
+    try:
+        new_book.save()
+    except BookError:
+        print('Sorry ' + new_book.title + ' is already saved in the list.')
+        main()
```

Comment:
I'm gonna tell you what clara told me when I first solved the "Program crashes when modifying book that is not in database" issue. ("Why does (new_book.save()) raise an exception? Investigate and fix, instead of adding try-except here")